test/WebSites/ModelBindingWebSite/Models/ProductValidatorAttibute.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+
+namespace ModelBindingWebSite
+{
+    public class ProductValidatorAttribute : ValidationAttribute
+    {
+        protected override ValidationResult IsValid(object value, ValidationContext validationContext)
+        {
+            var objectType = value.GetType();
+
+            var neededProperties =
+              objectType.GetProperties()
+              .Where(propertyInfo => propertyInfo.Name == "Name" || propertyInfo.Name == "Country")
+              .ToArray();
```

Comment:
similar to the admin validator in the `TryValidateModel` PR, conditionally cast to `Product`.  fail if that returns `null`.  then will be able to use the properties directly.